realm/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -91,6 +97,73 @@ public static void prepareDatabaseFromAssets(Context context, String realmPath,
         return key;
     }
 
+    /**
+     * Returns a Logger that will fail if it is asked to log a message above a certain level.
+     *
+     * @param failureLevel {@link Log} level from which the unit test will fail.
+     * @return Logger implementation
+     */
+    public static Logger getFailureLogger(final int failureLevel) {
+        return new Logger() {
+
+            private void failIfEqualOrAbove(int logLevel, int failureLevel) {
+                if (logLevel >= failureLevel) {
+                    fail("Message logged that was above valid level: " + logLevel + " >= " + failureLevel);
```

Comment:
I think throwing exception is better at this level. One can call this method inside a thread that doesn't catch `AssertionError` & get nothing reported, whereas a checked exception force the caller to handle it.